### PR TITLE
Remove GGP dependencies from CDC RSync

### PR DESCRIPTION
**Pull request by ljusten**
A bunch of CLs that make CDC RSync more independent of GGP.

### CODE REVIEW

File: asset_stream_manager/session.cc [hunk 1]
```diff
@@ -51,7 +51,7 @@ Session::Session(std::string instance_id, std::string instance_ip,
                    /*forward_output_to_logging=*/true),
       metrics_recorder_(std::move(metrics_recorder)) {
   assert(metrics_recorder_);
-  remote_util_.SetIpAndPort(instance_ip, instance_port);
+  remote_util_.SetHostAndPort(instance_ip, instance_port);
```
Comment (chrschng) [inline asset_stream_manager/session.cc:54]: Can you change `instance_ip` to `instance_host` to match the semantic?

Comment (ljusten) [reply asset_stream_manager/session.cc:54]: I'll do this in a follow-up CL. I didn't want to touch AS30 yet unless necessary.

File: asset_stream_manager/cdc_fuse_manager.cc [hunk 2]
```diff
@@ -29,7 +29,7 @@ namespace {
 constexpr char kFuseFilename[] = "cdc_fuse_fs";
 constexpr char kLibFuseFilename[] = "libfuse.so";
 constexpr char kFuseStdoutPrefix[] = "cdc_fuse_fs_stdout";
-constexpr char kRemoteToolsBinDir[] = "/opt/developer/tools/bin/";
+constexpr char kRemoteToolsBinDir[] = "~/.cache/cdc_file_transfer/";
```
Comment (chrschng) [inline asset_stream_manager/cdc_fuse_manager.cc:32]: Note that `~` is expanded by the shell running on the instance. That should work when using SCP, but there can be situations in which `$HOME` works but `~` does not. If it works right now, then keep it.

File: cdc_rsync/cdc_rsync.h [hunk 3]
```diff
@@ -46,7 +46,9 @@ struct Options {
   const char* copy_dest = nullptr;
   int compress_level = 6;
   int connection_timeout_sec = 10;
-
+  const char* ssh_command = nullptr;
```
Comment (chrschng) [inline cdc_rsync/cdc_rsync.h]: Is there a reason to not use `absl::string_view`?

File: cdc_rsync/cdc_rsync_client.cc [hunk 4]
```diff
@@ -109,24 +109,26 @@ GgpRsyncClient::GgpRsyncClient(const Options& options, PathFilter path_filter,
...
 
 absl::Status GgpRsyncClient::Run() {
-  absl::Status status = remote_util_.GetInitStatus();
-  if (!status.ok()) {
-    return WrapStatus(status, "Failed to initialize critical components");
-  }
-
   // Initialize |remote_util_|.
-  remote_util_.SetIpAndPort(options_.ip, options_.port);
+  remote_util_.SetHostAndPort(options_.ip, options_.port);
```
Comment (chrschng) [inline cdc_rsync/cdc_rsync_client.cc]: Can you change `Options::ip` to `Options::host`?

File: common/remote_util.cc [hunk 5]
```diff
@@ -69,9 +71,16 @@ RemoteUtil::RemoteUtil(int verbosity, bool quiet,
...
 
-void RemoteUtil::SetIpAndPort(const std::string& gamelet_ip, int ssh_port) {
-  gamelet_ip_ = gamelet_ip;
-  ssh_port_ = ssh_port;
+void RemoteUtil::SetHostAndPort(std::string hostname, int port) {
+  hostname_ = std::move(hostname);
+  ssh_port_ = port;
+}
+void RemoteUtil::SetScpCommand(std::string scp_command) {
+  scp_command_ = scp_command;
```
Comment (chrschng) [inline common/remote_util.cc:49]: Nit: Use `std::move`.

File: common/remote_util.cc [hunk 6]
```diff
@@ -69,9 +71,16 @@ RemoteUtil::RemoteUtil(int verbosity, bool quiet,
...
+}
+
+void RemoteUtil::SetSshCommand(std::string ssh_command) {
+  ssh_command_ = ssh_command;
```
Comment (chrschng) [inline common/remote_util.cc:53]: Nit: Use `std::move`.

File: common/remote_util.cc [hunk 7]
```diff
@@ -123,9 +125,9 @@ absl::Status RemoteUtil::Sync(std::vector<std::string> source_filepaths,
 
   ProcessStartInfo start_info;
   start_info.command = absl::StrFormat(
-      "%s --ip=%s --port=%i -z %s %s%s",
-      path::Join(sdk_util_.GetDevBinPath(), "cdc_rsync"),
-      QuoteArgument(gamelet_ip_), ssh_port_,
+      "cdc_rsync.exe --ip=%s --port=%i -z "
```
Comment (chrschng) [inline common/remote_util.cc:98]: Can you drop the `.exe` extension? Windows doesn't need it and it's one less change to make if we want to port it to Linux.

File: common/remote_util.h [hunk 8]
```diff
@@ -22,14 +22,15 @@
...
 #include "common/process.h"
-#include "common/sdk_util.h"
 
 namespace cdc_ft {
 
 // Utilities for executing remote commands on a gamelet through SSH.
 // Windows-only.
 class RemoteUtil {
  public:
+   const int kDefaultSshPort = 22;
```
Comment (chrschng) [inline common/remote_util.h]: Nit: `constexpr`

File: common/remote_util.h [hunk 9]
```diff
@@ -38,44 +39,48 @@ class RemoteUtil {
...
              bool forward_output_to_log);
 
-  // Returns the initialization status. Should be OK unless in case of some rare
-  // internal error. Should be checked before accessing any members.
-  const absl::Status& GetInitStatus() const {
-    return sdk_util_.GetInitStatus();
-  }
+  // Sets the hostname of the remote instance and the ssh tunnel port. The
+  // hostname can be an IP address and/or contain a username, e.g. user@1.2.3.4.
+  void SetHostAndPort(std::string hostname, int ssh_port);
```
Comment (chrschng) [inline common/remote_util.h:44]: Can you change this to any of `Set(Remote|Target|Dest)HostAndPort` to make the direction clearer?

File: common/remote_util.h [hunk 10]
```diff
@@ -101,7 +106,7 @@ class RemoteUtil {
   bool Quiet() const { return quiet_; }
 
  private:
-  // Verifies that both |gamelet_ip_| and |ssh_port_| are set.
+  // Verifies that both |hostname_| and |ssh_port_| are set.
   absl::Status CheckIpPort();
```
Comment (chrschng) [inline common/remote_util.h:131]: Nit: `CheckHostPort()`

File: common/remote_util.h [hunk 11]
```diff
@@ -113,9 +118,10 @@ class RemoteUtil {
   ProcessFactory* const process_factory_;
   const bool forward_output_to_log_;
 
-  SdkUtil sdk_util_;
-  std::string gamelet_ip_;
-  int ssh_port_ = 0;
+  std::string scp_command_ = "scp.exe";
```
Comment (chrschng) [inline common/remote_util.h:142]: Again, can you drop the `.exe` suffix to make it easier to port to Linux?

File: common/sdk_util.cc [hunk 12]
```diff
@@ -31,13 +31,6 @@ SdkUtil::SdkUtil() {
   absl::Status status = path::GetEnv("GGP_SDK_PATH", &ggp_sdk_path_env_);
   if (absl::IsNotFound(status) || ggp_sdk_path_env_.empty())
     ggp_sdk_path_env_ = path::Join(program_files_path_, "GGP SDK");
```
Comment (chrschng) [inline common/sdk_util.cc:33]: I wonder if we should change this, maybe just leave it empty. Thoughts?

File: README.md [hunk 13]
```diff
@@ -11,4 +11,54 @@ Tool to sync files to a remote machine, similar to the standard Linux
 uses a higher performing remote diffing approach based on CDC.
```
Comment (chrschng) [inline README.md:11]: Can you change the leading words of this paragraph (line 9) to form a complete sentence?

File: asset_stream_manager/cdc_fuse_manager.cc (hunk 2 above)
Comment (ljusten) [reply asset_stream_manager/cdc_fuse_manager.cc:32]: Yeah, I think for that reason I had to remove quoting from remote_util.cc. IIRC I used wordexp at some point in the server, but since I don't see that anymore, I guess I have removed it again. Anyway, this works for both Stadia and gLinux.

File: cdc_rsync/cdc_rsync_client.cc (hunk 4 above)
Comment (ljusten) [reply cdc_rsync/cdc_rsync_client.cc]: I'm calling this user_host in a later CL in this chain.

File: cdc_rsync/cdc_rsync.h (hunk 3 above)
Comment (ljusten) [reply cdc_rsync/cdc_rsync.h]: In this incarnation, the data is passed through a DLL boundary, so one needs to use POD types. absl::string_view is probably aright, but probably also not a good idea, e.g. if two different ABSL versions would be used between DLLs and they changed something about the type. Anyway, in a later CL I removed the DLL and used std::strings, which is necessary since some vars can come from environment vars (CDC_SSH_COMMAND).

File: common/remote_util.cc (hunk 6 above)
Comment (ljusten) [reply common/remote_util.cc:53]: Done

File: common/remote_util.cc (hunk 5 above)
Comment (ljusten) [reply common/remote_util.cc:49]: Done.

File: cdc_rsync_cli/params.cc [hunk 14]
```diff
@@ -73,11 +73,24 @@ For partially matching files only the deltas are transferred.
     --copy-dest dir       Use files from dir as sync base if files are missing
     --ssh-command         Path and arguments of SSH command to use, e.g.
                           C:\path\to\ssh.exe -F config -i id_rsa -oStrictHostKeyChecking=yes -oUserKnownHostsFile="""known_hosts"""
+                          Can aslo be specified by the CDC_SSH_COMMAND environment variable.
     --scp-command         Path and arguments of SSH command to use, e.g.
                           C:\path\to\scp.exe -F config -i id_rsa -oStrictHostKeyChecking=yes -oUserKnownHostsFile="""known_hosts"""
+                          Can aslo be specified by the CDC_SCP_COMMAND environment variable.
```
Comment (chrschng) [inline cdc_rsync_cli/params.cc:81]: Typo: Oslo

**Review by chrschng — CHANGES_REQUESTED** [review]
Done with all commits in this stack.

File: asset_stream_manager/cdc_fuse_manager.cc (hunk 2 above)
Comment (ljusten) [reply asset_stream_manager/cdc_fuse_manager.cc:32]: That being said, it would make sense to run through some corner cases later and/or adapt the e2e tests to make sure everything still works.

File: common/remote_util.cc (hunk 7 above)
Comment (ljusten) [reply common/remote_util.cc:98]: Done. I'm actually planning to remove this code since it's only used to deploy FUSE in debug mode, which was mainly used by Lena as her home internet wasn't so good. For us scp is probably even faster.

File: common/remote_util.h (hunk 8 above)
Comment (ljusten) [reply common/remote_util.h]: Already done in a later CL.

File: common/remote_util.h (hunk 10 above)
Comment (ljusten) [reply common/remote_util.h:131]: Done. Btw. since SetHostPort is now the only way to get it into a proper state, we might as well take that as constructor args.

File: common/remote_util.h (hunk 11 above)
Comment (ljusten) [reply common/remote_util.h:142]: Done.

File: cdc_rsync/cdc_rsync_client.cc [hunk 15]
```diff
@@ -190,7 +190,7 @@ absl::Status GgpRsyncClient::StartServer() {
 
   // Find available local and remote ports for port forwarding.
   absl::StatusOr<int> port_res =
-      port_manager_.ReservePort(options_.connection_timeout_sec);
+      port_manager_.ReservePort(false, options_.connection_timeout_sec);
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync/cdc_rsync_client.cc:191]: Done.

File: cdc_rsync/error_messages.h [hunk 16]
```diff
@@ -21,12 +21,12 @@ namespace cdc_ft {
 
 // Server connection timed out. SSH probably stale.
 constexpr char kMsgFmtConnectionTimeout[] =
-    "Server connection timed out. Please re-run 'ggp ssh init' and verify that "
-    "the IP '%s' and the port '%i' are correct.";
+    "Server connection timed out. Verify that the IP '%s' and the port '%i' "
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync/error_messages.h]: Already changed in a later CL to
Server connection timed out. Verify that host '%s' and port '%i' are correct, or specify a larger timeout with --contimeout.

File: cdc_rsync/error_messages.h [hunk 17]
```diff
@@ -21,12 +21,12 @@ namespace cdc_ft {
...
+    "are correct.";
 
 // Server connection timed out and IP was not passed in. Probably network error.
 constexpr char kMsgConnectionTimeoutWithIp[] =
-    "Server connection timed out. Please check your network connection.";
+    "Server connection timed out. Check your network connection.";
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync/error_messages.h]: https://developers.google.com/style/tone#politeness

But actually unused since we now always have an IP. Oh and the name was wrong, should've been kMsgConnectionTimeoutWithOUTIp.

File: README.md [hunk 18]
```diff
@@ -11,4 +11,54 @@ Tool to sync files to a remote machine, similar to the standard Linux
...
 
 ## Asset Streaming
-Tool to stream assets from a Windows machine to a Linux device.
\ No newline at end of file
+Tool to stream assets from a Windows machine to a Linux device.
```
[the remark this answers is not in the document]
Comment (ljusten) [reply README.md:14]: Done, though I prefer the slightly more concise "Tool to stream".

File: README.md [hunk 19]
```diff
@@ -11,4 +11,54 @@ Tool to sync files to a remote machine, similar to the standard Linux
...
+  ```
+* Copy the Linux build output files `cdc_fuse_fs` and `libfuse.so` from 
+  `bazel-bin/cdc_fuse_fs` on the Linux system to `bazel-bin\asset_stream_manager`
+  on the Windows machine.
+* Copy the Linux build output file `cdc_rsync_server` from 
+  `bazel-bin/cdc_rsync_server` on the Linux system to `bazel-bin\cdc_rsync_cli`
+  on the Windows machine.
+
+## Usage
+To copy the contents of the Windows directory `C:\path\to\assets` to
```
[the remark this answers is not in the document]
Comment (ljusten) [reply README.md:55]: Done. I've revamped the doc a little more and split up the Building section as well.

File: README.md (hunk 13 above)
Comment (ljusten) [reply README.md:11]: Done.

File: cdc_rsync/cdc_rsync.cc [hunk 20]
```diff
@@ -25,7 +25,8 @@
 namespace cdc_ft {
 namespace {
 
-ReturnCode TagToMessage(Tag tag, const Options* options, std::string* msg) {
+ReturnCode TagToMessage(Tag tag, const char* user_host, const Options* options,
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync/cdc_rsync.cc]: A later version passes in const cdc_ft::params::Parameters& params,.

File: cdc_rsync/cdc_rsync.cc [hunk 21]
```diff
@@ -69,7 +70,8 @@ PathFilter::Rule::Type ToInternalType(FilterRule::Type type) {
 ReturnCode Sync(const Options* options, const FilterRule* filter_rules,
                 size_t num_filter_rules, const char* sources_dir,
                 const char* const* sources, size_t num_sources,
-                const char* destination, const char** error_message) {
+                const char* user_host, const char* destination,
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync/cdc_rsync.cc]: I'm not 100% clear yet how we'lll finally do it. It would make sense to just take a target param that includes user@host:path (or even a URI). This would make sense if we eventually support local copies, but for now let's keep it this way.

File: cdc_rsync_cli/params.cc [hunk 22]
```diff
@@ -39,11 +40,13 @@ Synchronizes local files and files on a gamelet. Matching files are skipped.
...
 Usage:
-  cdc_rsync [options] source [source]... destination
+  cdc_rsync [options] source [source]... [user@]host:destination
 
 Parameters:
   source                  Local file or folder to be copied
-  destination             Destination folder on the gamelet
+  user                    Remote SSH user name
+  host                    Remote host or IP address
+  destination             Remote destination folder
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync_cli/params.cc:51]: Done.

File: cdc_rsync/cdc_rsync.cc [hunk 23]
```diff
@@ -69,7 +70,8 @@ PathFilter::Rule::Type ToInternalType(FilterRule::Type type) {
 ReturnCode Sync(const Options* options, const FilterRule* filter_rules,
                 size_t num_filter_rules, const char* sources_dir,
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync/cdc_rsync.cc]: This function will go away.

File: cdc_rsync_cli/params.cc [hunk 24]
```diff
@@ -324,6 +316,35 @@ bool CheckParameters(const Parameters& params, bool help) {
               << std::endl;
   }
 
+  if (params.sources.empty() && params.destination.empty()) {
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync_cli/params.cc:337]: It provides a slightly better error message. Otherwise, you could go

cdc_rsync --port=1234
-> Error: Missing destination
cdc_rsync --port=1234 /mnt/developer
-> Error: Missing source

File: cdc_rsync_cli/params.cc [hunk 25]
```diff
@@ -324,6 +316,35 @@ bool CheckParameters(const Parameters& params, bool help) {
...
+    PrintError("Missing destination");
+    return false;
+  }
+
+  if (params.sources.empty()) {
+    // If one arg was passed on the command line, it is not clear whether it
+    // was supposed to be a source or destination. Try to infer that, e.g.
+    //   cdc_rsync *.txt          -> Missing destination
+    //   cdc_rsync /mnt/developer -> Missing source
+    bool missing_src = params.destination[0] == '/';
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync_cli/params.cc:352]: OK, removed the heuristic.

File: cdc_rsync_cli/params.cc [hunk 26]
```diff
@@ -324,6 +316,35 @@ bool CheckParameters(const Parameters& params, bool help) {
...
+
+    PrintError("Missing %s", missing_src ? "source" : "destination");
+    return false;
+  }
+
+  if (params.user_host.empty()) {
+    PrintError(
+        "No remote host specified in destination '%s'. "
+        "Expected [user@]host:dest.",
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync_cli/params.cc:361]: Done.

File: cdc_rsync_cli/params.cc [hunk 27]
```diff
@@ -348,6 +369,25 @@ bool CheckOptionResult(OptionResult result, const std::string& name,
   return true;
 }
 
+// Removes the user/host part of |destination| and puts it into |user_host|,
+// e.g. if |destination| is initially "user@foo.com:~/file", it is "~/file"
+// afterward and |user_host| is |user@foo.com|. Does not touch Windows drives,
+// e.g. C:\foo.
+void ParseUserHost(std::string* destination, std::string* user_host) {
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync_cli/params.cc:394]: Done.

[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync_cli/params.cc:394]: Reason is I don't need it. Both ssh and scp take user@host, so no need to split, but since the destination is handled within our tool, I need to split that off.

File: common/remote_util.cc [hunk 28]
```diff
@@ -212,6 +182,26 @@ ProcessStartInfo RemoteUtil::BuildProcessStartInfoForSshInternal(
   return start_info;
 }
 
+std::string RemoteUtil::EscapeForWindows(const std::string& argument) {
+  std::string str =
+      std::regex_replace(argument, std::regex(R"(\\*(?=""|$))"), "$1$1");
+  return std::regex_replace(str, std::regex("\""), "\\\"");
```
[the remark this answers is not in the document]
Comment (ljusten) [reply common/remote_util.cc:188]: Done. Hope I didn't mess it up.

File: common/remote_util.cc [hunk 29]
```diff
@@ -212,6 +182,26 @@ ProcessStartInfo RemoteUtil::BuildProcessStartInfoForSshInternal(
   return start_info;
 }
 
+std::string RemoteUtil::EscapeForWindows(const std::string& argument) {
+  std::string str =
+      std::regex_replace(argument, std::regex(R"(\\*(?=""|$))"), "$1$1");
```
[the remark this answers is not in the document]
Comment (ljusten) [reply common/remote_util.cc:187]: Done (in the .h file)

File: common/remote_util.cc [hunk 30]
```diff
@@ -212,6 +182,26 @@ ProcessStartInfo RemoteUtil::BuildProcessStartInfoForSshInternal(
...
+  return std::regex_replace(str, std::regex("\""), "\\\"");
+}
+
+std::string RemoteUtil::QuoteArgument(const std::string& argument) {
+  return absl::StrFormat("\"%s\"", EscapeForWindows(argument));
```
[the remark this answers is not in the document]
Comment (ljusten) [reply common/remote_util.cc:192]: Done.

File: cdc_rsync_cli/params.cc [hunk 31]
```diff
@@ -80,7 +80,7 @@ For partially matching files only the deltas are transferred.
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync_cli/params.cc:84]: I forgot the details, but some escaping nightmare. Outer escapes are for Windows, then it needs another layer of escaped quotes.

File: cdc_rsync_cli/params.cc [hunk 32]
```diff
@@ -73,11 +73,24 @@ For partially matching files only the deltas are transferred.
     --copy-dest dir       Use files from dir as sync base if files are missing
     --ssh-command         Path and arguments of SSH command to use, e.g.
                           C:\path\to\ssh.exe -F config -i id_rsa -oStrictHostKeyChecking=yes -oUserKnownHostsFile="""known_hosts"""
+                          Can aslo be specified by the CDC_SSH_COMMAND environment variable.
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync_cli/params.cc:78]: Done.

File: cdc_rsync/main.cc [hunk 33]
```diff
@@ -18,13 +18,93 @@
...
+      *msg = kMsgAddressInUse;
+      return ReturnCode::kAddressInUse;
 
+    case cdc_ft::Tag::kDeployServer:
+      *msg = kMsgDeployFailed;
+      return ReturnCode::kDeployFailed;
+
+    case cdc_ft::Tag::kConnectionTimeout:
+      *msg = absl::StrFormat(kMsgFmtConnectionTimeout, params.user_host,
+                             params.options.port);
```
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync/main.cc:94]: Done.

File: cdc_rsync/error_messages.h (hunk 16 above)
[the remark this answers is not in the document]
Comment (ljusten) [reply cdc_rsync/error_messages.h]: Google asks us not to be polite: https://developers.google.com/style/tone#politeness

File: common/port_manager.h [hunk 34]
```diff
@@ -51,10 +51,13 @@ class PortManager {
   // Reserves a port in the range passed to the constructor. The port is
   // released automatically upon destruction if ReleasePort() is not called
   // explicitly.
-  // |timeout_sec| is the timeout for finding available ports on the gamelet
-  // instance. Returns a DeadlineExceeded error if the timeout is exceeded.
+  // |check_remote| determines whether the remote port should be checked as
+  // well. If false, the check is skipped and a port might be returned that is
```
[the remark this answers is not in the document]
Comment (ljusten) [reply common/port_manager.h:55]: I'm done with

File: common/sdk_util.cc (hunk 12 above)
Comment (ljusten) [reply common/sdk_util.cc:33]: My plan is to get rid of SdkUtil entirely. Right now, there are still a few usages
- GetServicesConfigPath() by AS30 for the config.json file -> Plan is to either remove it and use Absl's ability to specify a flags file, or to keep it and provide flag to specify the config file. I'll probably keep it since it's less work (still referenced by tests, for instance).
- GetLogPath() by AS30. Will provide a flag to set the log path, and log to stdout by default.
- GetDevBinPath() by LocalAssetsStreamManagerServiceImpl to find ggp.exe. This could just be removed if we rely on ggp being in PATH. Also, this code path is Stadia specific, so we could remove it or move the logic to find the full path to LocalAssetsStreamManagerServiceImpl entirely.

File: common/remote_util.h (hunk 9 above)
Comment (ljusten) [reply common/remote_util.h:44]: Renamed to SetUserHostAndPort() for consistency.

File: README.md [hunk 35]
```diff
@@ -6,14 +6,40 @@ on Content Defined Chunking (CDC), in particular
 to split up files into chunks.
 
 ## CDC RSync
-Tool to sync files to a remote machine, similar to the standard Linux
-[rsync](https://linux.die.net/man/1/rsync). It supports fast compression and
-uses a higher performing remote diffing approach based on CDC.
 
-## Asset Streaming
-Tool to stream assets from a Windows machine to a Linux device.
+CDC RSync is a ool to sync files to a remote machine, similar to the standard
```
Comment (chrschng) [inline README.md:10]: ypo

Comment (chrschng) [inline README.md:10]: I think it is important to mention that it currently only supports a Windows -> Linux sync.

**Comment by chrschng** [comment]
Thanks!

**Review by chrschng — APPROVED** [review]

File: README.md (hunk 35 above)
Comment (ljusten) [reply README.md:10]: one.

Comment (ljusten) [reply README.md]: Done.

**Review by chrschng — APPROVED** [review]
Thanks again!
